src/rules/no-deprecated-functions.ts

```diff
@@ -1,9 +1,65 @@
+import { JSONSchemaForNPMPackageJsonFiles } from '@schemastore/package';
 import {
   AST_NODE_TYPES,
   TSESTree,
 } from '@typescript-eslint/experimental-utils';
 import { createRule, getNodeName } from './utils';
 
+interface ContextSettings {
+  jest?: EslintPluginJestSettings;
+}
+
+export type JestVersion =
+  | 14
+  | 15
+  | 16
+  | 17
+  | 18
+  | 19
+  | 20
+  | 21
+  | 22
+  | 23
+  | 24
+  | 25
+  | 26
+  | 27
+  | number;
+
+interface EslintPluginJestSettings {
+  version: JestVersion;
+}
+
+let cachedJestVersion: JestVersion | null = null;
+
+/** @internal */
+export const _clearCachedJestVersion = () => (cachedJestVersion = null);
+
+const detectJestVersion = (): JestVersion => {
+  if (cachedJestVersion) {
+    return cachedJestVersion;
+  }
+
+  try {
+    const jestPath = require.resolve('jest/package.json', {
+      paths: [process.cwd()],
+    });
+
+    // eslint-disable-next-line @typescript-eslint/no-require-imports
+    const jestPackageJson = require(jestPath) as JSONSchemaForNPMPackageJsonFiles;
+
+    if (jestPackageJson.version) {
+      const [majorVersion] = jestPackageJson.version.split('.');
+
+      return (cachedJestVersion = parseInt(majorVersion, 10));
+    }
+  } catch {}
+
+  throw new Error(
```

Comment:
@SimenB I just remembered: I originally used `toThrowErrorMatchingInlineSnapshot`, but it kept complaining the snapshot was different everytime due to missing indentation at the start & end.

I'll try and put it into a reproduction 🐛